theseus/core/tests/test_cost_function.py

```diff
@@ -115,23 +116,27 @@ def error_fn(optim_vars, aux_vars):
                 )
         else:
             # check that the error function returns the correct value
-            cost_function = th.AutoDiffCostFunction(
-                optim_vars,
-                error_fn,
-                err_dim,
-                cost_weight=cost_weight,
-                aux_vars=aux_vars,
-            )
-            err = cost_function.error()
-            assert err.allclose(variable_values.sum() * torch.ones(batch_size, err_dim))
+            for batched in [True, False]:
```

Comment:
Is the code in this and other tests exactly the same as the previous version except for the `batched` loop? If so, I think a clearer change would be to do 

```python
@pytest.mark.parametrize("batched", [True, False])
def test_autodiff_cost_function_error_and_jacobians_shape(batched):
...
```

That would make the diff easier to read, since the only change would be the top line of the function. Right now it's harder to follow because of the change in indentation between the two versions.